cpp/src/gandiva/precompiled/decimal_sql.h

```diff
@@ -0,0 +1,34 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Can you explain why this is called `decimal_sql.h`? Does it have anything to do with SQL?

Reply:
i was initially thinking we may want to support decimal ops where the sql rules but scale/precision don't matter (eg. like python add). but, probably we can get to it when required.

for now, i have removed the 'sql' from the names.